dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/io/readers/SubSnpCoreFieldsReader.java

```diff
@@ -0,0 +1,112 @@
+         CASE
+            WHEN loc.orientation = 1 THEN -1 ELSE 1
+         END AS snp_orientation,
+         CASE
+            WHEN ctg.orient = 1 THEN -1 ELSE 1
+         END AS contig_orientation
+     FROM
+         b148_snpcontigloc loc JOIN
+         b148_contiginfo ctg ON ctg.ctg_id = loc.ctg_id JOIN
+         snpsubsnplink link ON loc.snp_id = link.snp_id JOIN
+         subsnp sub ON link.subsnp_id = sub.subsnp_id
+     WHERE
+         ctg.group_term IN($extract_mappings_for)
+         AND ctg.group_label LIKE '$group_label'
+     ORDER BY sorting_id ASC;
+ */
+public class SubSnpCoreFieldsReader extends JdbcPagingItemReader<SubSnpCoreFields> {
+
+    public SubSnpCoreFieldsReader(String assembly, List<String> assemblyType, DataSource dataSource, int pageSize)
+            throws Exception {
+        if (pageSize < 1) {
+            throw new IllegalArgumentException("Page size must be greater than zero");
+        }
+
+        setDataSource(dataSource);
+        setQueryProvider(createQueryProvider(dataSource));
+        setRowMapper(new SubSnpCoreFieldsRowMapper());
+        setPageSize(pageSize);
+
+        Map<String, Object> parameterValues = new HashMap<>();
+        parameterValues.put("assemblyType", String.join(", ", assemblyType));
+        parameterValues.put("assembly", assembly);
+        setParameterValues(parameterValues);
+    }
+
+    private PagingQueryProvider createQueryProvider(DataSource dataSource) throws Exception {
+        SqlPagingQueryProviderFactoryBean factoryBean = new SqlPagingQueryProviderFactoryBean();
+        factoryBean.setDataSource(dataSource);
+        factoryBean.setSelectClause(
+                "SELECT " +
+                        "sub.subsnp_id AS ss_id, " +
+                        "loc.snp_id AS rs_id, " +
+                        "ctg.contig_name AS contig_name, " +
+                        "ctg.contig_gi AS contig_id, " +
+                        "loc.asn_from +1 AS contig_start, " +
+                        "loc.asn_to +1 as contig_end, " +
+                        "ctg.contig_chr AS chromosome, " +
+                        "loc.phys_pos_from + 1 AS chromosome_start, " +
+                        "loc.phys_pos_from + 1 + loc.asn_to - loc.asn_from AS chromosome_end, " +
+                        "CASE " +
+                        "   WHEN loc.orientation = 1 THEN -1 ELSE 1 " +
+                        "END AS snp_orientation, " +
+                        "CASE " +
+                        "   WHEN ctg.orient = 1 THEN -1 ELSE 1 " +
+                        "END AS contig_orientation"
+        );
+        factoryBean.setFromClause(
+                "FROM" +
+                        "   b148_snpcontigloc loc JOIN " +
+                        "   b148_contiginfo ctg ON ctg.ctg_id = loc.ctg_id JOIN " +
+                        "   snpsubsnplink link ON loc.snp_id = link.snp_id JOIN " +
+                        "   subsnp sub ON link.subsnp_id = sub.subsnp_id"
```

Comment:
I don't really know why are there extra spaces at the beginning of each line, here and in the WHERE clause...